src/main/webapp/ts/profileBuilder.ts

```diff
@@ -1,39 +1,66 @@
-type authInfoFromServlet = {
+type authInfo = {
     loginUrl: string;
     logoutUrl: string;
     hasProfile: boolean;
 };
 
 function setForm(value: string) {
-    const menteeForm =  document.getElementById('Mentee');
+    const menteeForm = document.getElementById('Mentee');
     const mentorForm = document.getElementById('Mentor');
     const userTypeButton = document.getElementById('user-input');
 
-    if(menteeForm && mentorForm && userTypeButton){
-        if(value == 'Mentee') {
-            menteeForm.setAttribute('style','display:block;');
-            mentorForm.setAttribute('style','display:none;');
-            userTypeButton.setAttribute('value','Mentee');
+    const menteeInputs:string[] = ['name-input','school-input', 'major-input' ];
+    const mentorInputs:string[] =['profName-input','company-input','careerTitle-input', 'bio-input', 'specialty-input']
+
+    if (menteeForm && mentorForm && userTypeButton) {
+        if (value == 'Mentee') {
+            menteeForm.setAttribute('style', 'display:block;');
+            mentorForm.setAttribute('style', 'display:none;');
+            userTypeButton.setAttribute('value', 'Mentee');
+
+            if (menteeInputs && mentorInputs) {
+                setRequiredInputs(menteeInputs, true);
+                setRequiredInputs(mentorInputs, false);
+            }
+            else {
+                console.error("Required forms failed for mentee form");
+            }
         }
         else {
-            mentorForm.setAttribute('style','display:block;');
-            menteeForm.setAttribute('style','display:none;');
-            userTypeButton.setAttribute('value','Mentor');
+            mentorForm.setAttribute('style', 'display:block;');
+            menteeForm.setAttribute('style', 'display:none;');
+            userTypeButton.setAttribute('value', 'Mentor');
+
+            if (mentorInputs && menteeInputs) {
+                setRequiredInputs(mentorInputs, true);
+                setRequiredInputs(menteeInputs, false);
+            }
+            else {
+                console.error("Required forms failed for mentor form");
+            }
         }
     }
-} 
+}
+
+function setRequiredInputs(elements: any, boolValue: boolean) {
```

Comment:
Can we work on choosing more descriptive variable names? Using vague variable names makes code harder for readers to understand.

elements is actually an id which is a string. 
boolValue is actually whether or not the element is required.

Reply:
Done